modules/p2/ding_list/ding_list.module

```diff
@@ -435,6 +435,11 @@ function ding_list_ctools_plugin_directory($owner, $plugin_type) {
  * loans list.
  */
 function ding_list_user_login(&$edit, $account) {
+  // Don't update users when openlist isn't available.
+  if (ding_provider_implements('openlist', 'is_available') && !ding_provider_invoke('openlist', 'is_available')) {
+    return;
+  }
```

Comment:
I wonder if this is the right way to go handle the issue in general.

As I understand it the problem is that exceptions are thrown and logged during installation [which we also see on Circle CI during site-install](https://circleci.com/gh/ding2/ding2/2256) because openlist is not configured.

This change tried to determine if openlist is available specifically for user login but why this specific case and not other attempts to access openlist?

If possible it would be preferable to  determine why user login is called during the installation phase or why  OpenList is accessed with the parameters instead.

Reply:
How about [`drupal_installation_attempted`](https://api.drupal.org/api/drupal/includes%21bootstrap.inc/function/drupal_installation_attempted/7.x)?

If we can use this here with a code comment explaining why this is necessary then the new provider hook is no longer needed and we should also be able to leave out the changes to `ting_openlist_request`.